data/systemd/snapd.manage

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+#
+# This script acts as an entry point for running snapd from snapd.snap on
+# a base snap that otherwise doesn't ship snapd itself. The entry point
+# arranges for appropriate system integration, in particular by creating
+# and starting systemd units for running snapd without any confinement.
+#
+# Arguments:
+# $1: start|stop
+# $2: prefix of the installed snapd snap (e.g. /snap/snapd/42/)
+#     (on a fedora base this might be e.g. /var/lib/snaps...)
+
+set -e
+
+CMD="$1"
+PREFIX="$2"
+
+UNITS="snapd.service snapd.autoimport.service snapd.core-fixup.service \
+       snapd.seeded.service snapd.snap-repair.service \
+       snapd.snap-repair.timer snapd.socket"
+ENABLE_UNITS="snapd.system-shutdown.service"
+
+# TODO: discuss ID_LIKE=ubuntu-core or ID_LIKE=snap-core ?
+if ! grep -q "ID=ubuntu-core" /etc/os-release && \
+        ! grep -q "ID_LIKE=ubuntu-core" /etc/os-release; then
+    echo "cannot run this helper on a non ubuntu-core system"
+    exit 1
+fi
+
+case $CMD in
+    start)
+        echo "Running start action"
+        if [ "$PREFIX" = "" ]; then
+            echo "cannot run without the prefix for the snapd snap"
+            exit 1
+        fi
+        # use existing snapd.service as a template and modify so that the
+        # right prefix is used
+        for u in $UNITS $ENABLE_UNITS; do
```

Comment:
Nitpick, lack of shellcheck markers makes me realize that this is not shellchecked at all. Can we ensure this is not under the radar as a critical part of the infrastructure.

Reply:
Good point, I added a make check rule to the Makefile which is run as part of the make all target.